android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt

```diff
@@ -396,39 +396,54 @@ class JsonConverter {
-            }
-            if (event is SourceEvent.AudioTrackChanged) {
-                json.putMap("oldAudioTrack", fromAudioTrack(event.oldAudioTrack))
-                json.putMap("newAudioTrack", fromAudioTrack(event.newAudioTrack))
-            }
-            if (event is SourceEvent.AudioTrackRemoved) {
-                json.putMap("audioTrack", fromAudioTrack(event.audioTrack))
-            }
-            if (event is SourceEvent.SubtitleTrackAdded) {
-                json.putMap("subtitleTrack", fromSubtitleTrack(event.subtitleTrack))
-            }
-            if (event is SourceEvent.SubtitleTrackRemoved) {
-                json.putMap("subtitleTrack", fromSubtitleTrack(event.subtitleTrack))
-            }
-            if (event is SourceEvent.SubtitleTrackChanged) {
-                json.putMap("oldSubtitleTrack", fromSubtitleTrack(event.oldSubtitleTrack))
-                json.putMap("newSubtitleTrack", fromSubtitleTrack(event.newSubtitleTrack))
+            when (event) {
+                is SourceEvent.Load -> {
+                    json.putMap("source", fromSource(event.source))
+                }
+
+                is SourceEvent.Loaded -> {
+                    json.putMap("source", fromSource(event.source))
+                }
+
+                is SourceEvent.Error -> {
+                    json.putInt("code", event.code.value)
+                    json.putString("message", event.message)
+                }
+
+                is SourceEvent.Warning -> {
+                    json.putInt("code", event.code.value)
+                    json.putString("message", event.message)
+                }
+
+                is SourceEvent.AudioTrackAdded -> {
+                    json.putMap("audioTrack", fromAudioTrack(event.audioTrack))
+                }
+
+                is SourceEvent.AudioTrackChanged -> {
+                    json.putMap("oldAudioTrack", fromAudioTrack(event.oldAudioTrack))
+                    json.putMap("newAudioTrack", fromAudioTrack(event.newAudioTrack))
+                }
+
+                is SourceEvent.AudioTrackRemoved -> {
+                    json.putMap("audioTrack", fromAudioTrack(event.audioTrack))
+                }
+
+                is SourceEvent.SubtitleTrackAdded -> {
+                    json.putMap("subtitleTrack", fromSubtitleTrack(event.subtitleTrack))
+                }
+
+                is SourceEvent.SubtitleTrackRemoved -> {
+                    json.putMap("subtitleTrack", fromSubtitleTrack(event.subtitleTrack))
+                }
+
+                is SourceEvent.SubtitleTrackChanged -> {
+                    json.putMap("oldSubtitleTrack", fromSubtitleTrack(event.oldSubtitleTrack))
+                    json.putMap("newSubtitleTrack", fromSubtitleTrack(event.newSubtitleTrack))
+                }
+
+                else -> {
+                    // Event is not supported yet or does not have any additional data
```

Comment:
A `Log` would be nice :)

Reply:
I don't think it's a good idea to log here as there are many events that just don't have any additional data that would suddenly fill up the logs even though they are fully supported. 
In order to have a meaningful log we would need to explicitly include those in the 'when' entry as supported, but I consider this out of scope for this quick improvement.

Reply:
Makes sense!